File: design-pattern-java/app/src/main/java/design/pattern/java/patterns/structural/facade/ThirdPartyLibraries.java
package design.pattern.java.patterns.structural.facade;

public class ThirdPartyLibraries {
  public static class VideoFile {
    public final String filename;

    public VideoFile(String filename) {
      this.filename = filename;
    }
  }

  public static class CompressionCodec {
  }

  public static class OggCompressionCodec extends CompressionCodec {
  }

  public static class MPEG4CompressionCodec extends CompressionCodec {
  }

  public static class CodecFactory {
    public static String extract(VideoFile file) {
      return "CodecFactory.extract: ".concat(file.filename);
    }
  }

  public static class BitrateReader {
    static String read(String value, String source) {
      return "BitReader.read: ".concat(value).concat(",").concat(source);
    }

    static String buffer(String value, CompressionCodec compression) {
      return "BitReader.buffer: ".concat(value);
    }
  }

  public static class AudioMixer {
    public String fix(String value) {
      return "AudioMixer.fix: ".concat(value);
    }
  }

}

File: design-pattern-jvm/app/src/main/kotlin/design/pattern/kotlin/patterns/structural/facade/ThirdPartyLibraries.kt
package design.pattern.kotlin.patterns.structural.facade

class ThirdPartyLibraries {
    class VideoFile(val filename: String)
    open class CompressionCodec
    class OggCompressionCodec : CompressionCodec()
    class MPEG4CompressionCodec : CompressionCodec()
    object CodecFactory {
        fun extract(file: VideoFile): String {
            return "CodecFactory.extract: " + file.filename
        }
    }

    object BitrateReader {
        fun read(value: String, source: String): String {
            return "BitReader.read: $value,$source"
        }

        fun buffer(value: String, compression: CompressionCodec?): String {
            return "BitReader.buffer: $value"
        }
    }

    class AudioMixer {
        fun fix(value: String): String {
            return "AudioMixer.fix: $value"
        }
    }
}